mlir_graphblas/algorithms.py

```diff
@@ -168,7 +170,8 @@ def dense_neural_network_combined(
             bias_matrix_csr = irb.util.ptr8_to_tensor(
                 bias_matrix_ptr, "tensor<?x?xf64, #CSR64>"
             )
-            bias_matrix = irb.util.cast_csr_to_csc(bias_matrix_csr)
+            bias_matrix_csx = irb.util.cast_csr_to_csx(bias_matrix_csr)
+            bias_matrix_csc = irb.util.cast_csx_to_csc(bias_matrix_csx)
```

Comment:
Ideally, we could handle these cast calls in the PtrToTensorOp directly.
```python
bias_matrix_csc = irb.util.ptr8_to_tensor(bias_matrix_ptr, "tensor<?x?xf64, #CSC64>"
```
The op would create the `ret_val`, inspect its type and realize we need to do some casting. The op would then inject those casting lines automatically.

Reply:
The changes in https://github.com/metagraph-dev/mlir-graphblas/pull/84/commits/c188c91127553f940b228133029247c2951fa0cd implement this suggestion.